src/tools/run-make-support/src/lib.rs

```diff
@@ -208,6 +208,17 @@ pub fn bin_name(name: &str) -> String {
     if is_windows() { format!("{name}.exe") } else { name.to_string() }
 }
 
+/// Count the number of files in the current
+/// working directory possessing the extension `ext`.
+#[track_caller]
+pub fn count_files_with_extension(ext: &str) -> usize {
+    fs_wrapper::read_dir(cwd())
+        .filter_map(|entry| entry.ok())
+        .filter(|entry| entry.path().is_file())
+        .filter(|entry| entry.path().extension().unwrap() == ext)
+        .count()
+}
```

Comment:
Suggestion: I would recommend building this on top of `walkdir` also like in https://github.com/rust-lang/rust/pull/126427#discussion_r1643325225 so we don't swallow errors and if we do encounter errors, we report something that's actually meaningful for the tester.

Reply:
@rustbot author